src/twisted/trial/test/__init__.py

```diff
@@ -4,3 +4,39 @@
 """
 Unit tests for the Trial unit-testing framework.
 """
+
+from hypothesis import HealthCheck, settings
+
+
+def _activateHypothesisProfile():
+    """
+    Load a Hypothesis profile appropriate for a Twisted test suite.
+    """
+    deterministic = settings(
+        # Disable the deadline.  It is too hard to guarantee that a particular
+        # piece of Python code will always run in less than some fixed amount
+        # of time.  Hardware capabilities, the OS scheduler, the Python
+        # garbage collector, and other factors all combine to make substantial
+        # outliers possible.  Such failures are a distraction from development
+        # and a hassle on continuous integration environments.
+        deadline=None,
+        suppress_health_check=[
+            # With the same reasoning as above, disable the Hypothesis time
+            # limit on data generation by example search strategies.
+            HealthCheck.too_slow,
```

Comment:
I guess that this is ok for now.

My understanding is that we will not implement any custom data generation.

But, maybe instead of unlimited time, we can add something like 1 minute.

I guess the limit is big enough for a test input.

The idea is that if something goes wrong on the data generation, you should get a better error message, rather than seeing the whole CI cancelled due to timeout.

Reply:
I'm still very skeptical of applying any limit.  Basically, our CI providers give us no guarantees whatsoever how much processor time our jobs will get.  Chances are most of the time they'll give us plenty and Hypothesis won't have a problem - but how close to "all of the time" will that be?  I dunno.

Or I guess I could look at it from a different perspective.  _trial_ imposes a 2 minute timeout per Deferred-returning test, by default ... So imposing a 2 minute timeout on Hypothesis stuff is no worse than that.  I guess people probably haven't been complaining lately about the 2 minute trial timeout spuriously engaging on CI?

Reply:
Ah, but there is no way to set a custom amount of time that defines "too slow".  You can either have the "too slow" check or you can turn it off.  The "deadline", which applies to how long the test method itself takes, is configurable, so I could still apply a timeout to that.  However, I don't think this will interrupt a test that takes too long, it will just cause it to fail afterwards.  So for the case of "make CI fail with a better report" I don't think it helps - whether there's a deadline or not, if the test runs for long enough that CI decides to kill it, CI's just gonna kill it and you get no extra info.

So I guess I'll leave this alone for now.  Maybe we'll figure out a better configuration at some future point.

Reply:
Sure. Sounds right. If we can't get better error reporting, then don't bother here.

I don't remember seeing any CI error due to trial 2 minute limit.

And on my local run, if a test takes more than 10 seconds, I kill it :)